opentracing-api/src/main/java/io/opentracing/ActiveSpan.java

```diff
@@ -0,0 +1,71 @@
+package io.opentracing;
+
+import java.io.Closeable;
+
+/**
+ * In any execution context (or any thread, etc), there is at most one "active" {@link Span} primarily responsible for
+ * the work accomplished by the surrounding application code. That active Span may be accessed via the
+ * {@link ActiveSpanSource#active()} method. If the application needs to defer work that should be part of the same Span, the
+ * Source provides a {@link ActiveSpan#defer} method that returns a {@link Continuation}; this continuation may be used
+ * to re-activate and continue the {@link Span} in that other asynchronous executor and/or thread.
+ *
+ * <p>
+ * {@link ActiveSpan}s are created via {@link Tracer.SpanBuilder#startAndActivate()} or {@link ActiveSpanSource#adopt}. They can
+ * be {@link ActiveSpan#defer()}ed as {@link ActiveSpan.Continuation}s, then re-{@link Continuation#activate()}d later.
+ *
+ * @see ActiveSpanSource
+ */
+public interface ActiveSpan extends Closeable, Span {
+    /**
+     * Mark the end of the active period for the {@link Span} pinned by this {@link ActiveSpan}. When the last
+     * {@link ActiveSpan} is deactivated for a given {@link Span}, it is automatically {@link Span#finish()}ed.
+     * <p>
+     * <p>
+     * NOTE: It is an error to call deactivate() more than once on a single {@link ActiveSpan}.
+     *
+     * @see Closeable#close() {@link ActiveSpan}s are auto-closeable and may be used in try-with-resources blocks
+     */
+    void deactivate();
```

Comment:
I am not sure what to say about this... I definitely see your point and agree that there is a risk here. Personally, I'm ok with it, but it's unfortunately just a judgement call and I have no data to suggest that programmers won't trip over this.

I basically want to get people out of the habit of calling `finish` and get in the habit of using `try {}` for most straightforward Span lifetimes.

So, to be clear: you would be happier if we wrote something like this in the spec/docs:

```
If `Span.finish` is called while `ActiveSpan`s are still extant, the effect is to turn all future `deactivate` calls into noops.
```

Correct? I am not convinced I want to put that sort of bookkeeping responsibility upon implementations (esp since it still creates a weird ambiguity about the tagging/logging features of the Span after finish has been called), but maybe we can agree on what the hypothetical change would be.